Earthfile

```diff
@@ -0,0 +1,257 @@
+    COMMAND
+    RUN __install make bash gcc g++ unzip curl tar gzip git musl-dev \
+                  linux-headers openssl-dev python3
+
+# Environment targets are defined below. These do not have build outputs, but
+# are rather themselves the "outputs" to be used as the environment for subsequent
+# tasks
+
+env.c6:
+    # A CentOS 6 environment.
+    FROM +init --base=docker.io/library/centos:6
+    DO +CENTOS6_SETUP
+
+env.c7:
+    # A CentOS 7 environment.
+    FROM +init --base=docker.io/library/centos:7
+    DO +REDHAT_SETUP
+
+env.rl8:
+    # CentOS 8 is cancelled. Use RockyLinux 8 for our RHEL 8 environment.
+    FROM +init --base=docker.io/library/rockylinux:8
+    DO +REDHAT_SETUP
+
+# Utility command for Ubuntu environments
+ENV_UBUNTU:
+    COMMAND
+    ARG --required version
+    FROM +init --base=docker.io/library/ubuntu:$version
+    DO +DEBIAN_SETUP
+
+env.u14:
+    # An Ubuntu 14.04 environment
+    DO +ENV_UBUNTU --version 14.04
+
+env.u16:
+    # An Ubuntu 16.04 environment
+    DO +ENV_UBUNTU --version 16.04
+
+env.u18:
+    # An Ubuntu 18.04 environment
+    DO +ENV_UBUNTU --version 18.04
+
+env.u20:
+    # An Ubuntu 20.04 environment
+    DO +ENV_UBUNTU --version 20.04
+
+env.u22:
+    # An Ubuntu 22.04 environment
+    DO +ENV_UBUNTU --version 22.04
+
+env.amzn1:
+    # An Amazon "1" environment. (AmazonLinux 2018)
+    FROM +init --base=docker.io/library/amazonlinux:2018.03
+    DO +AMZ_SETUP
+
+env.amzn2:
+    # An AmazonLinux 2 environment
+    FROM +init --base=docker.io/library/amazonlinux:2
+    DO +AMZ_SETUP
+
+# Utility command for Debian setup
+ENV_DEBIAN:
+    COMMAND
+    ARG --required version
+    FROM +init --base=docker.io/library/debian:$version
+    DO +DEBIAN_SETUP
+
+env.deb9:
+    # A Debian 9.2 environment
+    DO +ENV_DEBIAN --version 9.2
+
+env.deb10:
+    # A Debian 10.0 environment
+    DO +ENV_DEBIAN --version 10.0
+
+env.deb11:
+    # A Debian 11.0 environment
+    DO +ENV_DEBIAN --version 11.0
+
+env.sles15:
+    # An OpenSUSE Leap 15.0 environment.
+    FROM +init --base=docker.io/opensuse/leap:15.0
+    DO +SLES_SETUP
+
+env.alpine:
+    FROM +init --base=docker.io/library/alpine:3.17
+    DO +ALPINE_SETUP
+
+# Utility: Warm-up obtaining CMake and Ninja for the build. This is usually
+# very quick, but on some platforms we need to compile them from source.
+CACHE_WARMUP:
+    COMMAND
+    # Copy only the scripts that are strictly necessary for the operation, to
+    # avoid cache invalidation later on.
+    COPY .evergreen/setup-env.sh \
+         .evergreen/init.sh \
+         .evergreen/ensure-cmake.sh \
+         .evergreen/ensure-ninja.sh \
+         /T/
+    RUN bash /T/ensure-cmake.sh
+    RUN env NINJA_EXE=/usr/local/bin/ninja \
+        bash /T/ensure-ninja.sh
+
+COPY_SOURCE:
+    COMMAND
+    COPY --dir \
+        .git/ \
+        cmake/ \
+        kms-message/ \
+        test/ \
+        debian/ \
+        src/ \
+        etc/ \
+        LICENSE \
+        VERSION_CURRENT \
```

Comment:
```suggestion
```

VERSION_CURRENT is generated by cmake. Running `earthly +build --env=u22` may result in `Error: File not found "/VERSION_CURRENT"` if libmongocrypt was not previously configured in the build context.

Reply:
Good point. Although this does present an issue on e.g. CentOS 6, where the system Git and Python cannot generate the `VERSION_CURRENT` file correctly. A problem for another time, perhaps.